service/service.go

```diff
@@ -108,6 +89,60 @@ func (svc *Service) Init(ctx context.Context, cfg *config.Config, buildTime, git
 	return nil
 }
 
+// initClients initializes external service clients based on the service configuration.
+// It sets up clients for Zebedee, DatasetAPI, and Topic services if their corresponding flags are enabled.
+func (svc *Service) initClients(ctx context.Context) {
+	if svc.Cfg.EnableZebedeeCallbacks {
+		svc.ZebedeeCli = GetZebedee(svc.Cfg)
+	} else {
+		log.Info(ctx, "healthchecks skipped for zebedee as callbacks disabled")
+	}
+
+	if svc.Cfg.EnableDatasetAPICallbacks {
+		svc.DatasetCli = GetDatasetClient(svc.Cfg)
+	} else {
+		log.Info(ctx, "healthchecks skipped for Dataset as callbacks disabled")
+	}
+
+	svc.TopicCli = GetTopicClient(svc.Cfg)
+}
+
+func (svc *Service) initConsumers(ctx context.Context) error {
+	var err error
+
+	if svc.Cfg.EnableZebedeeCallbacks || svc.Cfg.EnableDatasetAPICallbacks {
```

Comment:
Check for nil config here too. Also for `EnableSearchContentUpdatedHandler` too.